tools/serverpod_cli/lib/src/config/config.dart

```diff
@@ -152,41 +153,33 @@ class GeneratorConfig {
   /// All the modules including my self and internal modules.
   List<ModuleConfig> get modulesAll => _modules;
 
-  /// Create a new [GeneratorConfig] by loading the configuration in the [dir].
-  static Future<GeneratorConfig> load([String dir = '']) async {
-    var serverPackageDirectoryPathParts = p.split(dir);
+  /// Create a new [GeneratorConfig] by loading the configuration in the [serverRootDir].
+  static Future<GeneratorConfig> load([String serverRootDir = '']) async {
+    var serverPackageDirectoryPathParts = p.split(serverRootDir);
 
     Pubspec? pubspec;
     try {
-      pubspec = parsePubspec(File(p.join(dir, 'pubspec.yaml')));
+      pubspec = parsePubspec(File(p.join(serverRootDir, 'pubspec.yaml')));
     } catch (_) {}
 
     if (pubspec == null) {
-      log.error(
-        'Failed to load pubspec.yaml. Are you running serverpod from your '
-        'projects root directory?',
-      );
-
       throw const ServerpodProjectNotFoundException(
-        'Failed to load valid pubspec.yaml',
+        'Failed to load pubspec.yaml. Are you running serverpod from your '
+        'projects server root directory?',
       );
```

Comment:
Confusing with two error messages, where one is never surfaced. Thought it was clearer to just have a single error message.